### ISSUE TITLE
ModelOptimizationToolkit 

### ISSUE THREAD

**Issue by yahuuu** [tensorflow/tensorflow#51562]
### 1. System information

- quantized integer tflite model in Window10(without Nvidia GPU)
- TensorFlow installation method pip installed tensorflow-gpu-2.2.2 and tensorflow-2.2.2.
- try to run quantized model in mobile phone (Mi 9 with built-in GPU and DSP).
- MI 9: android 9
- nnapi-delegate tool version: 2.5.0
- gpu-delegate tool version:   2.5.0
- android.tools.build:gradle:    4.1.3
- compile Sdk version : 29 android 10
- build tools version: 29.0.2
- NDK version: 21.1.6352462

### 2. Code
`
    # part of my script
    try:
        keras_model = tf.keras.models.load_model(save_model_dir)
        # Normalize the input image so that each pixel value is between 0 to 1.
        converter = tf.lite.TFLiteConverter.from_keras_model(keras_model)
        converter.optimizations = [tf.lite.Optimize.DEFAULT]
        if INTEGER_QUANTIZATON:
            converter.representative_dataset = representative_data_gen1
            # Ensure that if any ops can't be quantized, the converter throws an error
            converter.representative_dataset = tf.lite.RepresentativeDataset(representative_data_gen1)
            converter.target_spec.supported_ops = [tf.lite.OpsSet.TFLITE_BUILTINS_INT8]
            # Set the input and output tensors to uint8 (APIs added in r2.3)
            converter.inference_input_type = tf.int8
            converter.inference_output_type = tf.int8
        tflite_model_quant = converter.convert()
        open(quantized_save_path, "wb").write(tflite_model_quant)
`

#### Option A: Reference colab notebooks

1)  Reference url:
https://www.tensorflow.org/lite/performance/post_training_integer_quant
```


### 3. Failure after conversion
I try to quantify the model to increase the speed of inference. The float32 tflite model work well on mobile gpu(mi 9) and DSP(mi 9), BUT afer integer quantized to int8 model, the int8 tflite inference slow down on gpu and cannot work on DSP. 
floate32 tflite model
size: 86 KB
after quantized to integer model : 
size: 38.3KB  

### 5. Android studio logs
2021-08-19 11:15:54.428 24724-24724/? A/DEBUG:     #61 pc 00000000000cfce8  /system/lib64/libart.so (art::ArtMethod::Invoke(art::Thread*, unsigned int*, unsigned int, art::JValue*, char const*)+232)
2021-08-19 11:15:54.428 24724-24724/? A/DEBUG:     #62 pc 000000000045dd7c  /system/lib64/libart.so (art::(anonymous namespace)::InvokeWithArgArray(art::ScopedObjectAccessAlreadyRunnable const&, art::ArtMethod*, art::(anonymous namespace)::ArgArray*, art::JValue*, char const*)+104)
2021-08-19 11:15:54.428 24724-24724/? A/DEBUG:     #63 pc 000000000045d9dc  /system/lib64/libart.so (art::InvokeWithVarArgs(art::ScopedObjectAccessAlreadyRunnable const&, _jobject*, _jmethodID*, std::__va_list)+424)
2021-08-19 11:15:54.428 24724-24724/? A/DEBUG:     #64 pc 0000000000362cd8  /system/lib64/libart.so (art::JNI::CallStaticVoidMethodV(_JNIEnv*, _jclass*, _jmethodID*, std::__va_list)+652)
2021-08-19 11:15:54.428 24724-24724/? A/DEBUG:     #65 pc 00000000000b2884  /system/lib64/libandroid_runtime.so (_JNIEnv::CallStaticVoidMethod(_jclass*, _jmethodID*, ...)+116)
2021-08-19 11:15:54.428 24724-24724/? A/DEBUG:     #66 pc 00000000000b53f4  /system/lib64/libandroid_runtime.so (android::AndroidRuntime::start(char const*, android::Vector<android::String8> const&, bool)+924)
2021-08-19 11:15:54.428 24724-24724/? A/DEBUG:     #67 pc 0000000000002528  /system/bin/app_process64 (main+2012)
2021-08-19 11:15:54.428 24724-24724/? A/DEBUG:     #68 pc 00000000000c888c  /system/lib64/libc.so (__libc_init+88)
2021-08-19 11:15:54.619 24724-24724/? E/crash_dump64: cannot open libmiuindbg.so: No such file or directory
2021-08-19 11:15:54.631 985-985/? E//system/bin/tombstoned: Tombstone written to: /data/tombstones/tombstone_04
2021-08-19 11:15:54.690 1410-1941/? E/InputDispatcher: channel 'b92a91a com.example.srdemo/com.example.srdemo.MainActivity (server)' ~ Channel is unrecoverably broken and will be disposed!
2021-08-19 11:15:54.706 930-930/? E/Camera2Client: notifyError: Error condition 0 reported by HAL, requestId -1
2021-08-19 11:15:54.742 783-2114/? E/BufferQueueProducer: [SurfaceView - com.example.srdemo/com.example.srdemo.MainActivity#1] queueBuffer: BufferQueue has been abandoned
2021-08-19 11:15:54.742 930-1840/? E/Surface: queueBuffer: error queuing buffer to SurfaceTexture, -19
2021-08-19 11:15:54.755 783-2114/? E/BufferQueueProducer: [SurfaceView - com.example.srdemo/com.example.srdemo.MainActivity#1] cancelBuffer: BufferQueue has been abandoned
2021-08-19 11:15:54.785 783-2114/? E/BufferQueueProducer: [SurfaceView - com.example.srdemo/com.example.srdemo.MainActivity#1] cancelBuffer: BufferQueue has been abandoned
2021-08-19 11:15:54.786 783-2114/? E/BufferQueueProducer: [SurfaceView - com.example.srdemo/com.example.srdemo.MainActivity#1] cancelBuffer: BufferQueue has been abandoned
2021-08-19 11:15:54.788 783-871/? E/BufferQueueProducer: [SurfaceView - com.example.srdemo/com.example.srdemo.MainActivity#1] cancelBuffer: BufferQueue has been abandoned
2021-08-19 11:15:54.789 783-871/? E/BufferQueueProducer: [SurfaceView - com.example.srdemo/com.example.srdemo.MainActivity#1] cancelBuffer: BufferQueue has been abandoned
2021-08-19 11:16:00.520 726-1976/? E/android.hardware.usb@1.0-service: uevent received SUBSYSTEM=dual_role_usb
![tmp1](https://user-images.githubusercontent.com/41911152/130003229-543ae47d-3d93-4968-b4ef-ce2631a77886.JPG)
![tmp2](https://user-images.githubusercontent.com/41911152/130003232-0b0f0353-cea0-400b-892c-a476ab7ad86b.JPG)

**Comment by yahuuu** [comment]
![tmp1](https://user-images.githubusercontent.com/41911152/130003560-1ff4c925-6810-4897-b806-b18efad6718c.JPG)
![tmp2](https://user-images.githubusercontent.com/41911152/130003564-04493290-1365-4bee-803b-e7737fe76fd7.JPG)

**Comment by mohantym** [comment]
Hi@ yahuuu !
Could you please try on latest stable version of TF 2.6 and let us know if this is still an issue.Thanks!
